Code/Tools/AssetProcessor/AssetBuilderSDK/AssetBuilderSDK/AssetBuilderSDK.h

```diff
@@ -850,6 +850,38 @@ namespace AssetBuilderSDK
         ProcessJobResponse m_response;
     };
 
+    //////////////////////////////////////////////////////////////////////////
+    struct BuilderRegistration
+    {
+        AZ_CLASS_ALLOCATOR(BuilderRegistration, AZ::OSAllocator, 0);
+        AZ_TYPE_INFO(BuilderRegistration, "{36E785C3-5046-4568-870A-336C8249E453}");
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        AZStd::string m_name;
+        AZStd::vector<AssetBuilderSDK::AssetBuilderPattern> m_patterns;
+        AZ::Uuid m_busId;
+        int m_version = 0;
+        AZStd::string m_analysisFingerprint;
+        AZ::u8 m_flags = 0;
+        AZStd::unordered_map<AZStd::string, AZ::u8> m_flagsByJobKey;
+        AZStd::unordered_map<AZStd::string, AZStd::unordered_set<AZ::u32>> m_productsToKeepOnFailure;
+    };
+
```

Comment:
Could this be moved way furtther down or below the fold in terms of private API?  we don't actually want folks to use this in their builders, right?

Longer term, its probably best if we moved the stuff that AP and builders used to communicate with each other (privately) into a private shared staticlib, and left the AssetBuilderSDK contianing only things that people making new builders need to implement ...

Reply:
I've updated the PR to move this type and all other AP-AssetBuilder specific communication types into an AssetBuilder static lib